main/src/library/Result.flix

```diff
@@ -66,9 +66,9 @@ namespace Result {
     /// Returns `Ok(v)` if `r1` is `Ok(v)`. Otherwise returns `r2`.
     ///
     @Time(1) @Space(1)
-    pub def withDefault(r1: Result[t, e1], r2: Result[t, e2]): Result[t, e2] = match r1 {
+    pub def withDefault(r2: {r2 :: Result[t, e2]}, r1: Result[t, e1]): Result[t, e2] = match r1 {
```

Comment:
I would suggest the following renames:

- `r2 -> default`
- `r1` -> `r`

Reply:
That way your pipeline is:

```
foo |> Result.withDefault(default = Ok("Blargh)") |> ...
```

Reply:
it is a good idea however default seems to be a reserved word

Reply:
Hmmm. I think it should be only a soft keyword (and we are planning to deprecated it anyway).

Reply:
Does it not parse correctly? That could be an issue!

Reply:
Instead of `default` we could call it `alternative`? Or something else? But then should the function be renamed? Tricky.

Reply:
Yes I have the following compile time error: `unused formal parameter`.
We maybe can name it `orElse` such as in Scala, however I think this is not appropriate to make that kind of choice inside this PR. I can't name it `def` since it is a reserved word used for function definition. `alternative` is a good idea however it not seems to be very natural to me.